sdk-core-protos/src/lib.rs

```diff
@@ -209,6 +210,24 @@ pub mod coresdk {
         }
 
         impl ActivityResolution {
+            /// A helper function for cases when you want to extract an activity's payload if it
+            /// completed successfully or return an error for all other outcomes. It's essentially
+            /// an alternative to [`unwrap_ok_payload`][`Self::unwrap_ok_payload`] that uses neither
+            /// `unwrap()` nor `panic!`.
```

Comment:
```suggestion
            /// Extract an activity's payload if it completed successfully, or return an error for all 
            /// other outcomes.
```

Can be a bit more succinct here